types/react-relay/lib/relay-experimental/EntryPointTypes.d.ts

```diff
@@ -1,83 +1,72 @@
+import { ComponentType } from 'react';
 import {
-    OperationType,
+    CacheConfig,
     ConcreteRequest,
-    RequestParameters,
+    DisposeFn,
+    GraphQLResponse,
     IEnvironment,
     Observable,
-    GraphQLResponse,
-    CacheConfig,
+    OperationType,
+    RequestParameters,
+    VariablesOf,
 } from 'relay-runtime';
+import { GetEntryPointComponentFromEntryPoint, GetEntryPointParamsFromEntryPoint } from './helpers';
 
-import { Component } from 'react';
+export { VariablesOf } from 'relay-runtime';
 
-export type PreloadFetchPolicy = 'store-or-network' | 'store-and-network' | 'network-only';
+export interface JSResourceReference<TModule> {
+    getModuleId(): string;
 
-export interface PreloadOptions {
-    readonly fetchKey?: string | number;
-    readonly fetchPolicy?: PreloadFetchPolicy;
-    readonly networkCacheConfig?: CacheConfig;
+    getModuleIfRequired(): TModule | null;
+
+    load(): Promise<TModule>;
 }
 
-export interface LoadQueryOptions {
-    fetchPolicy?: PreloadFetchPolicy;
+export type PreloadFetchPolicy = 'store-or-network' | 'store-and-network' | 'network-only';
+
+export type PreloadOptions = Readonly<{
+    fetchKey?: string | number;
+    fetchPolicy?: PreloadFetchPolicy | null;
+    networkCacheConfig?: CacheConfig | null;
+}>;
+
+export type LoadQueryOptions = Readonly<{
+    fetchPolicy?: PreloadFetchPolicy | null;
     networkCacheConfig?: CacheConfig | null;
     onQueryAstLoadTimeout?: (() => void) | null;
-}
+}>;
 
 // Note: the phantom type parameter here helps ensures that the
 // $Parameters.js value matches the type param provided to preloadQuery.
 // tslint:disable-next-line interface-over-type-literal
-export type PreloadableConcreteRequest<TQuery extends OperationType> = {};
-
-export interface EnvironmentProviderOptions {
-    [key: string]: unknown;
-}
-
-export type PreloadedQuery<
-    TQuery extends OperationType,
-    TEnvironmentProviderOptions = EnvironmentProviderOptions
-> =
-    | PreloadedQueryInner_DEPRECATED<TQuery, TEnvironmentProviderOptions>
-    | PreloadedQueryInner<TQuery, TEnvironmentProviderOptions>;
+export type PreloadableConcreteRequest<TQuery extends OperationType> = {
+    kind: 'PreloadableConcreteRequest';
+    params: RequestParameters;
+};
 
-export type PreloadedQueryInner_DEPRECATED<
```

Comment:
I got rid of this  _DEPRECATED stuff. Its basically what used to be sent through `prepareEntrypoint` which isnt even exported in real relay anymore. TS got tricky introducing this union, so please agree we cool to remove it.

cc @alloy